src/Raven.Client/ServerWide/DatabaseTopology.cs

```diff
@@ -442,29 +442,51 @@ public void RemoveFromTopology(string delDbFromNode)
         public string WhoseTaskIsIt(
             RachisState state,
             IDatabaseTask task,
-            Func<string> getLastResponsibleNode = null)
+            Func<string> getLastResponsibleNode = null,
+            List<string> explanations = null)
         {
             if (state == RachisState.Candidate || state == RachisState.Passive)
+            {
+                explanations?.Add($"Task doesn't belong to this node because the node is in '{state}' state");
                 return null;
+            }
+
+            explanations?.Add($"Node state is '{state}'");
 
             var mentorNode = task.GetMentorNode();
             if (mentorNode != null)
             {
                 if (Members.Contains(mentorNode))
+                {
+                    explanations?.Add($"Task belongs to mentor node ({mentorNode})");
                     return mentorNode;
-                
+                }
+
                 if (AllNodes.Contains(mentorNode) && task.IsPinnedToMentorNode())
+                {
+                    explanations?.Add($"Task is pinned to mentor node ({mentorNode})");
                     return mentorNode;
+                }
+
+                explanations?.Add($"Mentor node is {mentorNode}");
             }
 
             var lastResponsibleNode = getLastResponsibleNode?.Invoke();
             if (lastResponsibleNode != null)
+            {
+                explanations?.Add($"Task belongs to last responsible node ({lastResponsibleNode})");
                 return lastResponsibleNode;
+            }
+
+            var node = WhoseTaskIsIt(task, explanations);
+
+            if (node != null)
+                explanations?.Add($"Task belongs to node {node}");
```

Comment:
maybe we want here to lag that node is null as well?

Reply:
done